### PR TITLE
[SQL] `az sql mi create`: Add `--dns-zone-partner` optional parameter

### PR DESCRIPTION
**Pull request by jovancevic123**
**Related command**
az sql mi create

**Description**
This change is being made to supports creation of Geo-Secondary instance that needs to have same dns zone as it's partner in failover group(Geo-Primary) in order to enable creation of failover group.

**Testing Guide**

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
SQL

File: src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py [hunk 1]
```diff
@@ -5362,6 +5362,41 @@ def test_sql_managed_instance_mgmt(self, mi, rg):
         # test list sql managed_instance in the subscription should be at least 1
         self.cmd('sql mi list', checks=[JMESPathCheckGreaterThan('length(@)', 0)])
 
+    @AllowLargeResponse()
+    def test_sql_managed_instance_create(self):
+        # Values of existing resources in order to test this feature
+        location = 'eastus2euap'
+        resource_group = 'sqlmigeodr'
+        subnet = ('/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/sqlmigeodr/providers/'
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py:5370]:
```suggestion
        subnet = ('/subscriptions/self.get_subscription_id()/resourceGroups/sqlmigeodr/providers/'
```

File: src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py [hunk 2]
```diff
@@ -5362,6 +5362,41 @@ def test_sql_managed_instance_mgmt(self, mi, rg):
...
+
+        instance_name = self.create_random_name(managed_instance_name_prefix, managed_instance_name_max_length)
+        self.kwargs.update({
+            'loc': location,
+            'rg': resource_group,
+            'subnet': subnet,
+            'managed_instance_name': instance_name,
+            'username': 'admin123',
+            'admin_password': 'SecretPassword123',
+            'dns_zone_partner': '/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/kmatijevic-ha-testenv-canary/providers/Microsoft.Sql/managedInstances/ha-testenv-canary-gp-1'
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py:5381]:
```suggestion
            'dns_zone_partner': '/subscriptions/self.get_subscription_id()/resourceGroups/kmatijevic-ha-testenv-canary/providers/Microsoft.Sql/managedInstances/ha-testenv-canary-gp-1'
```

File: src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py [hunk 3]
```diff
@@ -5362,6 +5362,41 @@ def test_sql_managed_instance_mgmt(self, mi, rg):
         # test list sql managed_instance in the subscription should be at least 1
         self.cmd('sql mi list', checks=[JMESPathCheckGreaterThan('length(@)', 0)])
 
+    @AllowLargeResponse()
+    def test_sql_managed_instance_create(self):
+        # Values of existing resources in order to test this feature
```
[the remark this answers is not in the document]
Comment (jovancevic123) [reply src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py:5367]: Oh, yes. I forgot it. Thanks.

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**